src/test/java/core/basesyntax/storage/StrategySubtractionTest.java

```diff
@@ -0,0 +1,49 @@
+package core.basesyntax.storage;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.servise.FruitRecordDto;
+import core.basesyntax.servise.Operation;
+import core.basesyntax.servise.exception.InvalidValueOfAmountException;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class StrategySubtractionTest {
+    private static final Fruit BANANA = new Fruit("banana");
+    private static final Fruit APPLE = new Fruit("apple");
+    private Strategy subtraction;
+
+    @Before
+    public void setUp() {
+        Storage.storageOfFruits.put(BANANA, 25);
+        subtraction = new StrategySubtraction();
+    }
+
+    @Test
+    public void changeBalance_SubtractionAmount_Ok() {
+        FruitRecordDto firstTest = new FruitRecordDto(Operation.PURCHASE, BANANA, 25);
+        int actual = subtraction.changeBalance(firstTest);
+        int expected = 0;
+        assertEquals("Result was wrong! Expected: " + expected
+                + " but was: " + actual, expected, actual);
+    }
+
+    @Test (expected = InvalidValueOfAmountException.class)
+    public void changeBalance_SubtractionAmountLeseZero_NotOk() {
+        FruitRecordDto firstTest = new FruitRecordDto(Operation.PURCHASE, BANANA, 35);
```

Comment:
firstTest is not informative, let's try to improve this naming in all tests

Reply:
done